libminifi/include/capi/api.h

```diff
@@ -79,6 +79,13 @@ processor *add_processor_with_linkage(flow *flow, const char *processor_name);
 
 processor *add_python_processor(flow *, void (*ontrigger_callback)(processor_session *session));
 
+/**
+* Register your callback to received flow files that the flow failed to process
+* The flow file is deleted after the callback is executed, make sure to copy all the data you need!
+* The first callback should be registered before the flow is used. Can be changed later during runtime.
+*/
```

Comment:
what if failure is an optional condition? Why remove and not reuse? curious about comment above.

Reply:
Not sure I understand the Q properly. 
The current implementation only requires to add the callback before finalising the flow (as there is no way to add the callback processor later). Later the callback can be replaced.

Reply:
Sorry that was a bit terse -- I'm asking about the choice to remove the flow file by default. It seems that there is no choice to handle rolling back events ( to retry ) or keeping the flow file content for future operations without completely copying the data. That seems like it should be something that's optional

Reply:
^ again, happy to move this discussion to that same ticket referenced above and merge this. Seems like any of these ideas can be done after more discussion on this topic. Thanks!